### PR TITLE
Documentation for API Breaking Changes

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Add documentation explaining the ABI update rules to the ABI implementation code.

### CODE REVIEW

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 1]
```diff
@@ -1334,7 +1334,48 @@ static constexpr OrtApiBase ort_api_base = {
...
+	it to ort_api_1_to_2.
+
+	Then in GetApi we'd make it return ort_api_1_to_2 for versions 1 through 2.
+
+Second example, if we wanted to add and remove some members, we'd do this:
+
+	In include\onnxruntime\core\session\onnxruntime_c_api.h we'd make a copy of the OrtApi structure and name the
+	old one OrtApi1. In the new OrtApi we'd add or remove any members that we desire.
+
+	In this file, we'd create a new copy of ort_api_1 caled ort_api_2 and make the corresponding changes that were
```
Comment (pranavsharma) [inline onnxruntime/core/session/onnxruntime_c_api.cc]: typo: called

Comment (RyanUnderhill) [reply onnxruntime/core/session/onnxruntime_c_api.cc]: Good catch!

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 2]
```diff
@@ -1449,13 +1502,17 @@ static constexpr OrtApi ort_api_1 = {
...
 };
 
 ORT_API(const OrtApi*, OrtApis::GetApi, uint32_t version) {
-  if (version > 1)
-    return nullptr;
+  if (version >= 1 && version <= 2)
+    return &ort_api_1_to_2;
 
-  return &ort_api_1;
+  return nullptr;  // Unsupported version
```
Comment (pranavsharma) [inline onnxruntime/core/session/onnxruntime_c_api.cc]: Ryan and I discussed more. Nullptr doesn't tell the user if it's an unsupported version. I suggest we add a comment in the .h file stating this.

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 3]
```diff
@@ -1334,20 +1334,64 @@ static constexpr OrtApiBase ort_api_base = {
...
+	old one OrtApi1. In the new OrtApi we'd add or remove any members that we desire.
+
+	In this file, we'd create a new copy of ort_api_1_to_2 called ort_api_3 and make the corresponding changes that were
+	made to the new OrtApi.
+
+	In GetApi we now make it return ort_api_3 for version 3.
+*/
+
+static constexpr OrtApi ort_api_1_to_2 = {
+
```
Comment (pranavsharma) [inline onnxruntime/core/session/onnxruntime_c_api.cc]: Also, probably a better idea is to just state here that the ordering is important and only appends are allowed if using the same struct.

Comment (RyanUnderhill) [reply onnxruntime/core/session/onnxruntime_c_api.cc]: Good idea, I put in a comment. Hopefully it's clear

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 4]
```diff
@@ -1334,7 +1334,49 @@ static constexpr OrtApiBase ort_api_base = {
...
+	In this file, we'd correspondingly add the member values to the end of the ort_api_1_to_2 structure, and also rename
+	it to ort_api_1_to_3.
+
+	Then in GetApi we'd make it return ort_api_1_to_3 for versions 1 through 3.
+
+Second example, if we wanted to add and remove some members, we'd do this:
+
+	In include\onnxruntime\core\session\onnxruntime_c_api.h we'd make a copy of the OrtApi structure and name the
+	old one OrtApi1. In the new OrtApi we'd add or remove any members that we desire.
+
```
Comment (skottmckay) [inline onnxruntime/core/session/onnxruntime_c_api.cc]: OrtApi1 or OrtApi1to2 given it was handling both version right?

Comment (RyanUnderhill) [reply onnxruntime/core/session/onnxruntime_c_api.cc]: Yep! I wrote it originally without there being a version 2, so I missed that part.

**Pull request merged**